src/lib/services/lsp3-account.service.ts

```diff
@@ -259,31 +261,38 @@ export async function setData(
 
   const erc725Account = new UniversalProfile__factory(signer).attach(erc725AccountAddress);
 
-  let controllerAddress: string;
-  let signerPermissions: string;
+  let signersAddresses: string[];
+  let signersPermissions: string[];
 
-  if (typeof controllerAddresses[0] === 'string') {
-    controllerAddress = controllerAddresses[0];
-  } else {
-    controllerAddress = controllerAddresses[0].address;
-    signerPermissions = controllerAddresses[0].permissions;
+  if (controllerAddresses.every((address) => address as ControllerOptions)) {
+    controllerAddresses.map(
+      (controller, index) => (signersAddresses[index] = controller[index].address)
+    );
+    signersPermissions = controllerAddresses.map(
+      (controller, index) => controller[index].permissions
+    );
   }
 
   const keysToSet = [
     LSP3_UP_KEYS.UNIVERSAL_RECEIVER_DELEGATE_KEY,
-    PREFIX_PERMISSIONS + controllerAddress.substr(2), // TODO: handle multiple addresses,
-    PREFIX_PERMISSIONS + universalReceiverDelegateAddress.substr(2),
+    ...signersAddresses.map((key) => PREFIX_PERMISSIONS + key.substring(2)), // TODO: handle multiple addresses,
+    PREFIX_PERMISSIONS + universalReceiverDelegateAddress.substring(2),
     ADDRESS_PERMISSIONS_ARRAY_KEY,
-    ADDRESS_PERMISSIONS_ARRAY_KEY.slice(0, 34) + '00000000000000000000000000000000',
-    ADDRESS_PERMISSIONS_ARRAY_KEY.slice(0, 34) + '00000000000000000000000000000001',
+    ...signersAddresses.map(
```

Comment:
We can do this computation above and store in a variable name so it is clear what is being passed into the keys array